### ISSUE TITLE
Files.java uses Closer where try-with-resources would be enough

### ISSUE THREAD

**Issue by leventov** [google/guava#3173]
It seems to me that https://github.com/google/guava/blob/2cac83e70d77f0fa9b2352fe5ac994280fc3b028/guava/src/com/google/common/io/Files.java uses `Closer` at three places where try-with-resources statements would be enough.

**Comment by nikhilbarar** [comment]
Can i implement this?

**Comment by ronshapiro** [comment]
I believe the issue here is that try-with-resources is still not available on all versions of Android that we support. Is that right @cpovirk?

**Comment by JakeWharton** [comment]
It's on API 19+ natively or API 1+ with desugar/D8, which I'm not sure if you assume.

**Comment by leventov** [comment]
@ronshapiro but Android has a completely different source file https://github.com/google/guava/blob/master/android/guava/src/com/google/common/io/Files.java, isn't it?

**Comment by ronshapiro** [comment]
@leventov yes, but we try not to fork unless we feel a need. It adds extra work to maintain both.

@JakeWharton we don't yet at least

**Comment by cpovirk** [comment]
[We currently target API 15.](https://github.com/google/guava/blob/c2f8967beb5a6feb9b104739d453dc116982a76a/README.md#important-warnings) I don't have a good feel for whether it's safe to assume Desugar/D8. It hasn't been a priority to figure out, since it's almost entirely a question of whether we can use `try-with-resources` internally. We may want to figure that out as Desugar/D8 evolves to do more.
